src/crimson/osd/pg.cc

```diff
@@ -812,30 +812,31 @@ PG::with_head_obc(hobject_t oid, with_obc_func_t&& func)
         return load_head_obc(obc);
       });
     }
-    return loaded.then([func = std::move(func)](auto obc) {
+    return loaded.safe_then([func=std::move(func)](auto obc) {
       return func(std::move(obc));
     });
   });
 }
 
 template<RWState::State State>
-seastar::future<>
+PG::load_obc_ertr::future<>
 PG::with_clone_obc(hobject_t oid, with_obc_func_t&& func)
 {
   assert(!oid.is_head());
   return with_head_obc<RWState::RWREAD>(oid.get_head(),
-    [oid, func=std::move(func), this](auto head) {
+    [oid, func=std::move(func), this](auto head) -> load_obc_ertr::future<> {
     auto coid = resolve_oid(head->get_ro_ss(), oid);
     if (!coid) {
       // TODO: return crimson::ct_error::enoent::make();
       logger().error("with_clone_obc: {} clone not found", coid);
-      return seastar::make_ready_future<>();
+      return load_obc_ertr::make_ready_future<>();
     }
     auto [clone, existed] = shard_services.obc_registry.get_cached_obc(*coid);
     return clone->template with_lock<State>(
       [coid=*coid, existed=existed,
-      head=std::move(head), clone=std::move(clone), func=std::move(func), this] {
-      auto loaded = seastar::make_ready_future<ObjectContextRef>(clone);
+       head=std::move(head), clone=std::move(clone),
+       func=std::move(func), this]() -> load_obc_ertr::future<> {
```

Comment:
Do we need the `()` here?

Reply:
yes. otherwise i have
```
../src/crimson/osd/pg.cc: In lambda function:
../src/crimson/osd/pg.cc:838:36: error: expected ‘{’ before ‘->’ token
  838 |        func=std::move(func), this] -> load_obc_ertr::future<> {
      |                                    ^~
../src/crimson/osd/pg.cc: In lambda function:
../src/crimson/osd/pg.cc:838:39: error: invalid use of ‘using future = class crimson::errorator<crimson::unthrowable_wrapper<const std::error_code&, ((const std::error_code&)(& crimson::ec<std::errc::illegal_byte_sequence>))> >::_future<crimson::errorated_future_marker<v\
oid> >’
  838 |        func=std::move(func), this] -> load_obc_ertr::future<> {
      |                                       ^~~~~~~~~~~~~
```